### ISSUE TITLE
Linear probe pipeline is not fully reproducible across runs

### ISSUE THREAD

**Issue by kutayeroglu** [kutayeroglu/ijepa#82]
## Problem

`main_linprobe.py` is only partially deterministic. While the subset selection when using `train_frac` / `val_frac` is seeded (seeds 42 and 43 respectively in `src/datasets/singleGPU_imagenet1k.py`), the rest of the training pipeline has multiple unseeded sources of randomness. This means two runs with identical arguments can produce different accuracy numbers.

## Sources of non-determinism

1. **`SubsetRandomSampler` iteration order** — The sampler is constructed without a `generator` argument, so the per-epoch shuffling of selected indices uses the unseeded global PyTorch RNG. Batch composition differs across runs.

2. **Linear head weight initialization** — `nn.Linear` in `LinearProbeModel` uses PyTorch's default Kaiming init with no seed, so starting weights vary.

3. **Data augmentations** — `RandomResizedCrop` and `RandomHorizontalFlip` in the training transforms draw from the unseeded global RNG.

4. **CUDA non-determinism** — Neither `torch.backends.cudnn.deterministic` nor `torch.use_deterministic_algorithms()` is set, allowing cuDNN to select non-deterministic algorithms.

5. **Multi-worker DataLoader** — With `num_workers > 0`, each worker process gets its own RNG state. Without a `worker_init_fn`, augmentation randomness across workers is uncontrolled.

## What is already deterministic

- The **subset of indices** selected by `train_frac` (seed 42) and `val_frac` (seed 43) is reproducible. The same fraction always selects the same samples.

## Suggested fix

Add a `--seed` argument to `main_linprobe.py` and apply it globally at startup:

```python
import random
import numpy as np

def seed_everything(seed: int):
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False
```

Then thread the seed through to the remaining unseeded components:

- Pass a seeded `generator` to `SubsetRandomSampler`.
- Add a `worker_init_fn` to both `DataLoader` instances that seeds each worker deterministically (e.g., `base_seed + worker_id`).
- Optionally call `torch.use_deterministic_algorithms(True)` for strictest guarantees (may reduce performance).

## Impact

This mainly matters when comparing linear probe results across different pretrained checkpoints or data fractions — small accuracy differences could be due to random seed variance rather than genuine model quality differences. Fixing this would make such comparisons more reliable.
